stan/math/prim/eigen_plugins.h

```diff
@@ -6,33 +9,48 @@
  *
  * TODO(Andrew): Replace with std::void_t after move to C++17
  */
-template<class, class = void>
-struct is_fvar : std::false_type
+template <typename T, typename has_member<decltype(T::d_)>::type = 0>
+struct is_fvar : std::true_type
+{ };
```

Comment:
It needs to have the ```int``` return and default value to avoid the SFINAE issue.

You can see the failure with 8.3 here: https://godbolt.org/z/7vGjTM

Working with the int return here: https://godbolt.org/z/WTa78W

Reply:
Ah nice!